### PR TITLE
chore(TPGOV-584): JDBC metadata for a field

### PR DESCRIPTION
**Pull request by afourmi**
**What is the problem this Pull Request is trying to solve?**
 
**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-schema/daikon-dataset-schema/src/main/java/org/talend/daikon/schema/dataset/DatasetFieldSchema.java [hunk 1]
```diff
@@ -1,17 +1,13 @@
 package org.talend.daikon.schema.dataset;
 
-import com.fasterxml.jackson.annotation.JsonAnyGetter;
-import com.fasterxml.jackson.annotation.JsonAnySetter;
-import com.fasterxml.jackson.annotation.JsonFormat;
-import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.annotation.JsonUnwrapped;
+import com.fasterxml.jackson.annotation.*;
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-schema/src/main/java/org/talend/daikon/schema/dataset/DatasetFieldSchema.java:3]: please check your import configuration and avoid `.*` import

File: daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      "type": "string"
+    },
+    "key": {
+      "type": "boolean"
+    },
+    "foreignKey": {
+      "type": "boolean"
+    },
+    "unique": {
+      "type": "boolean"
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json:27]: can you please add a description for each defined field ?

File: daikon-schema/daikon-dataset-schema/src/test/java/org/talend/daikon/schema/DeserializerDatasetSchemaTest.java [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
-
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.junit.jupiter.api.Test;
 import org.talend.daikon.schema.dataset.DatasetSchema;
 import org.talend.daikon.schema.dataset.mapper.DatasetSchemaMapperConfiguration;
+import org.talend.daikon.schema.dataset.metadata.JDBCMetadata;
 
 import java.io.IOException;
 
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-schema/src/test/java/org/talend/daikon/schema/DeserializerDatasetSchemaTest.java:11]: same. Avoid `.*` importt

File: daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "$id": "https://org.talend.daikon/dataset.field.jdbc.metadata.schema.json",
+  "title": "Dataset field metadata",
+  "description": "The metadata for a field, coming from the original data source",
+  "type": "object",
+  "properties": {
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json:7]: Is there some mandatory field on a JDBC metadata object ?

File: daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "$id": "https://org.talend.daikon/dataset.field.jdbc.metadata.schema.json",
+  "title": "Dataset field metadata",
+  "description": "The metadata for a field, coming from the original data source",
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json:5]: Title and description should reflect that it is a JDBC Metadata field

File: daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json (hunk 4 above)
Comment (afourmi) [reply daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json:7]: no field is mandatory

File: daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json (hunk 2 above)
Comment (afourmi) [reply daikon-schema/daikon-dataset-schema/src/main/resources/dataset/datasetFieldJDBCMetadata.json]: sure, done

File: daikon-schema/daikon-dataset-schema/src/main/java/org/talend/daikon/schema/dataset/DatasetFieldSchema.java (hunk 1 above)
Comment (gabriel-cpereira) [reply daikon-schema/daikon-dataset-schema/src/main/java/org/talend/daikon/schema/dataset/DatasetFieldSchema.java:3]: Done

File: daikon-schema/daikon-dataset-schema/src/test/java/org/talend/daikon/schema/DeserializerDatasetSchemaTest.java (hunk 3 above)
Comment (gabriel-cpereira) [reply daikon-schema/daikon-dataset-schema/src/test/java/org/talend/daikon/schema/DeserializerDatasetSchemaTest.java]: Done

File: daikon-schema/daikon-dataset-schema/src/test/java/org/talend/daikon/schema/DeserializerDatasetSchemaTest.java [hunk 6]
```diff
@@ -1,16 +1,19 @@
...
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.junit.jupiter.api.Test;
 import org.talend.daikon.schema.dataset.DatasetSchema;
 import org.talend.daikon.schema.dataset.mapper.DatasetSchemaMapperConfiguration;
+import org.talend.daikon.schema.dataset.metadata.JDBCMetadata;
 
 import java.io.IOException;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
```
Comment (fabballe) [inline daikon-schema/daikon-dataset-schema/src/test/java/org/talend/daikon/schema/DeserializerDatasetSchemaTest.java:12]: static import should be set first. See Talend import configuration

File: daikon-schema/daikon-dataset-schema/src/main/java/org/talend/daikon/schema/dataset/metadata/JDBCMetadata.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+@Value
+@SuperBuilder(toBuilder = true)
+@Jacksonized
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeName("jdbcMetadata")
+public class JDBCMetadata extends AbstractOriginalFieldMetadata {
+
+    String type;
```
Comment (wwang-talend) [inline daikon-schema/daikon-dataset-schema/src/main/java/org/talend/daikon/schema/dataset/metadata/JDBCMetadata.java:18]: here the "type" mean the database origin data type? like VARCHAR, INT and so on?

**Comment by wwang-talend** [comment]
look ok for me except the pattern which we not set in cloud jdbc connector

File: daikon-schema/daikon-dataset-schema/src/main/java/org/talend/daikon/schema/dataset/metadata/JDBCMetadata.java (hunk 7 above)
Comment (wwang-talend) [reply daikon-schema/daikon-dataset-schema/src/main/java/org/talend/daikon/schema/dataset/metadata/JDBCMetadata.java:18]: i think so from the test case.

**Review by guiLeloup — APPROVED** [review]
Also LGTM 👍

**Pull request merged**